M2/Macaulay2/packages/HomologicalShiftIdeals.m2

```diff
@@ -0,0 +1,1001 @@
+      I = ideal(x_1^2,x_2^5,x_3,x_1*x_4);
+      supportIdeal(I)
+///);
+
+supportIdeal = method(TypicalValue=>List)
+supportIdeal (Ideal) := I -> (
+        if isMonomialIdeal(I)==false then return "expected a monomial ideal"
+        else S:=ring I;
+        n:=numgens S;  
+        GI:=flatten entries mingens I;
+        suppI:=set{};
+        for w in GI do (
+             s:=set support w;
+             suppI=suppI+s;
+        );
+        suppI
+    );
+
+-------------------------------------------------------------------------------------------
+-- Whether an ideal is fully supported
+-------------------------------------------------------------------------------------------
+mydoc = concatenate(mydoc,///
+Node
+  Key
+    isFullySupported
+    (isFullySupported,Ideal)
+  Headline
+    whether a monomial ideal is fully supported
+  Usage
+    isFullySupported(I)
+  Inputs
+    I:Ideal
+      a monomial ideal $I$ of a polynomial ring $S = K[x_1,\ldots,x_n]$, $K$ a field
+  Outputs
+    b:Boolean
+      whether $I$ is fully supported
+  Description
+    Text
+      A monomial ideal $I$ of a polynomial ring $S=K[x_1,\ldots,x_n]$ is fully supported if $\text{supp}(I)=\{x_1,\dots,x_n\}$.
+    Example
+      S = QQ[x_1..x_4];
+      I = ideal(x_1^2,x_2,x_3,x_1^2*x_4);
+      isFullySupported(I)
+    Example
+      S = QQ[x_1..x_4];
+      I = ideal(x_1^2,x_3,x_2*x_4);
+      isFullySupported(I)
+///);
+
+isFullySupported = method(TypicalValue=>Boolean)
+isFullySupported (Ideal) := I -> (
+        if isMonomialIdeal(I)==false then return "expected a monomial ideal"
+        else n:=numgens ring I;
+        suppI:=supportIdeal I;
+        if #suppI==n then return true
+        else return false
+    );
+
+-------------------------------------------------------------------------------------------
+--  Compute the monomial with a given multidegree
+-------------------------------------------------------------------------------------------
+mydoc = concatenate(mydoc,///
+Node
+  Key
+    toMonomial
+    (toMonomial,Ring,List)
+  Headline
+    compute the monomial with a given multidegree
+  Usage
+    toMonomial(S,l)
+  Inputs
+    S:Ring
+      a polynomial ring $S = K[x_1,\ldots,x_n]$, $K$ a field
+    l:List
+      the multidegree $(a_1,\dots,a_n)$
+  Outputs
+    u:RingElement
+      the monomial $x_1^{a_1}\cdots x_n^{a_n}\in S$
+  Description
+    Text
+      The multidegree of a monomial $u\in S=K[x_1,\ldots,x_n]$ is the vector $(a_1,a_2,\dots,a_n)$ where $a_i=\max\{j:x_i^j\ \text{divides}\ u\}$.
+    Example
+      S = QQ[x_1..x_4];
+      l = {0,2,1,2};
+      toMonomial(S,l)
+///);
+
+toMonomial = method(TypicalValue=>RingElement)
+toMonomial (Ring,List) := (S,l) -> (
+	n:=numgens S-1;
+	u:=1;
+        j:=0;
+        if not(#l==n+1)==true then return "expected a multidegree whose size equals the number of variables of the polynomial ring"
+        else for i from 0 to n do (
+            j=l#i;
+            u=u*S_i^j;
```

Comment:
Similar to my `sum` comment above, you can iteratively multiply using [product](https://macaulay2.com/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_product_lp__Visible__List_cm__Function_rp.html):

```m2
product(n + 1, i -> S_i^(l#i))
```